static/app/views/detectors/detectorListRow.tsx

```diff
@@ -0,0 +1,150 @@
+import styled from '@emotion/styled';
+
+import Checkbox from 'sentry/components/checkbox';
+import {Flex} from 'sentry/components/container/flex';
+import InteractionStateLayer from 'sentry/components/interactionStateLayer';
+import {
+  ConnectionCell,
+  type Item,
+} from 'sentry/components/workflowEngine/gridCell/connectionCell';
+import {EmptyCell} from 'sentry/components/workflowEngine/gridCell/emptyCell';
+import {IssueCell} from 'sentry/components/workflowEngine/gridCell/issueCell';
+import {NumberCell} from 'sentry/components/workflowEngine/gridCell/numberCell';
+import {TitleCell} from 'sentry/components/workflowEngine/gridCell/titleCell';
+import {tn} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {Group} from 'sentry/types/group';
+import type {AvatarProject} from 'sentry/types/project';
+
+export type Detector = {
+  automations: Item[];
+  groups: Group[];
+  id: string;
+  link: string;
+  name: string;
+  project: AvatarProject;
+  details?: string[];
+  disabled?: boolean;
+};
+
+type DetectorListRowProps = Detector & {
+  handleSelect: (id: string, checked: boolean) => void;
+  selected: boolean;
+};
+
+export function DetectorListRow({
+  automations,
+  groups,
+  id,
+  link,
+  name,
+  project,
+  details,
+  handleSelect,
+  selected,
+  disabled = false,
+}: DetectorListRowProps) {
+  return (
+    <RowWrapper disabled={disabled}>
+      <InteractionStateLayer />
+      <Flex style={{justifyContent: 'space-between'}}>
+        <Flex flex={1}>
+          <StyledCheckbox
+            checked={selected}
+            onChange={() => {
+              handleSelect(id, !selected);
+            }}
+          />
+          <StyledTitleCell
+            name={name}
+            project={project}
+            link={link}
+            details={details}
+            disabled={disabled}
+          />
+        </Flex>
+        <StyledEmptyCell />
+      </Flex>
+      <StyledIssueCell
+        {...(groups.length > 0 ? {group: groups[0]} : {})}
+        disabled={disabled}
+      />
+      <StyledNumberCell number={groups.length} />
+      <StyledConnectionCell
+        items={automations}
+        renderText={count => tn('%s automation', '%s automations', count)}
+      />
+    </RowWrapper>
+  );
+}
+
+const StyledCheckbox = styled(Checkbox)<{checked?: boolean}>`
+  align-self: center;
```

Comment:
If I don't center the checkbox the hover style looks weird and darkens the full container?

<img width="292" alt="Screenshot 2025-02-12 at 9 38 20 AM" src="https://github.com/user-attachments/assets/6e0b1044-07f9-4242-a68f-d7f7e19b412c" />

Reply:
Fixed this by adding `align-self: flex-start` 🎉